File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        currentSum = 0 
        maxSum = float('inf') * -1

        for i in range(len(nums)):
            currentSum = max(currentSum + nums[i], nums[i])
            
            if currentSum > maxSum:
                maxSum = currentSum

        return maxSum

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currSum = 0;
        int maxSum = nums[0];

        for (int i = 0; i < nums.length; i++){
            currSum = Math.max(nums[i], nums[i] + currSum);
            if(maxSum < currSum) maxSum = currSum;
        }

        return maxSum;
    }
}